src/common/framework/gpu/implementation.ts

```diff
@@ -1,19 +1,112 @@
 /// <reference types="@webgpu/types" />
 
-import { assert } from '../util/util.js';
+import * as fs from 'fs';
 
-let impl: GPU | undefined = undefined;
+let impl: Promise<GPU>;
 
-export function getGPU(): GPU {
+export class WebGPUWindow {
+  constructor({ width, height, title }: { width?: number; height?: number; title?: string }) {
+    module.exports = require('../../../../standalone/third_party/dawn/linux/index.node');
```

Comment:
I belive the idea is to mimic the syntax for WebGPU in-browser. One of the goals is to have completely portable code from browser to NodeJS and to standalone NAPI/JS engine implementations like KDS V8 DCP-evaluator.

Reply:
The module is put in global scope elsewhere though (`Object.assign(...)`), I'm not clear how `module.exports` is needed.

If I could get this running it would be easier to understand (I could just play with it).